android/authentication.md

```diff
@@ -1,615 +1,704 @@
 
-    ```
-    ❯ No, I will set up my own configuration.
-    ```
+```java
+AWSMobileClient.getInstance().initialize(getApplicationContext(), new Callback<UserStateDetails>() {
+     @Override
+    public void onResult(UserStateDetails userStateDetails) {
+        switch (userStateDetails.getUserState()){
+            case SIGNED_IN:
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        TextView textView = (TextView) findViewById(R.id.text);
+                        textView.setText("Logged IN");
+                    }
+                });
+                break;
+            case SIGNED_OUT:
+                textView.setText("Logged OUT");
+                break;
+            default:
+                AWSMobileClient.getInstance().signOut();
+                break;
+        }
+    }
 
-3. Choose to set up authentication flow using AWS IAM access controls.
+    @Override
+    public void onError(Exception e) {
+        Log.e("INIT", e.toString());
+    }
+});
+```
 
-    ```
-    ❯ User Sign-Up, Sign-In, connected with AWS IAM controls
-    ```
+You might leverage the above workflow to perform other actions in the `SIGNED_IN` case, such as calling [GraphQL or REST APIs with AWS AppSync and Amazon API Gateway](./api) or uploading content with [Amazon S3](./storage).
 
-4. Choose yes, to: `? Allow unauthenticated logins?`.
+## Guest access
 
-5. Choose yes, to: `? Do you want to enable 3rd party authentication providers in your identity pool?`.
+Many applications have UX with "Guest" or "Unauthenticated" users. This is provided out of the box with `AWSMobileClient` through the initialization routine you have added. However, the Amplify CLI does not enable this by default with the `amplify add auth` flow. You can enable this by running `amplify update auth` and choosing `No, I will setup my own configuration` when prompted. Ensure you choose the **...connected with AWS IAM controls** which will allow you to select **Allow unauthenticated logins**.
 
-6. Choose Facebook and then provide your Facebook app ID. To retrieve or create your Facebook app ID, see [Setting Up Facebook Authentication](http://docs.aws.amazon.com/aws-mobile/latest/developerguide/auth-facebook-setup.html).
+When complete run `amplify push` and your `awsconfiguration.json` will work automatically with your updated Cognito settings. The `AWSMobileClient` user session will automatically have permissions configured for Guest/Unauthenticated users upon initialization. 
 
-7. When configuration for Facebook sign-in is complete, the CLI displays a message confirming that you have configured local CLI metadata for this category. You can confirm this by viewing status.
+If you login in your app either using the [Drop-In Auth](#dropinui) or the [direct Auth APIs](#iosapis) then the `AWSMobileClient` user session will transition to an authenticated role.
```

Comment:
iosapis -> androidapis?

Reply:
This was conversion from the Margherita repo. Fixed